### ISSUE TITLE
A little bug when running softmax with debug version

### ISSUE THREAD

**Issue by koverlu** [ROCmSoftwarePlatform/MIOpen#10]
file: src/kernel_cache.cpp
line: 91
```c++
    std::cout << "runcl " << params << " src/Kernels/" << program_name << " -k " << kernel_name
              << " -dumpilisa -r 10"
              << " if#" << isize * 4 << ": if#" << msize * 4 << ": if#" << osize * 4 << ": iv#0 "
              << vgd[0] << "," << vgd[1] << "," << vgd[2] << "/" << vld[0] << "," << vld[1] << ","
              << vld[2] << std::endl;
```
When running command `MIOpenDriver softmax -n 64 -W 1 -H 1 -c 1000 -V 0 -i 1 -t 1`, with a debug building (NDBUG is undefined)
`vgd` will be a const vector which size is 1. (in src/ocl/softmaxocl.cpp#81)

**Comment by dagamayank** [comment]
@koverlu Thanks for reporting. We will fix it.

**Comment by dagamayank** [comment]
Fixed in 173fe4d350e40a669e86e10f1cb72d354e62f553
